packages/internal/src/priority-senders.js

```diff
@@ -0,0 +1,85 @@
+import { E, Far } from '@endo/far';
+
+const { Fail, quote: q } = assert;
+
+export const PRIORITY_SENDERS_NAMESPACE_RE = /^[a-zA-Z0-9_-]{1,50}$/;
+
+/**
+ *
+ * @param {ERef<import('./lib-chainStorage.js').StorageNode>} sendersNode
+ */
+export const makePrioritySendersManager = sendersNode => {
+  /**
+   * address to tuple with storage node and set of namespaces that requested priority
+   *
+   * @type {Map<string, readonly [node: StorageNode, namespaces: Set<string>]>}
+   */
+  const addressRecords = new Map();
+
+  const refreshVstorage = (node, namespaces) => {
+    E(node).setValue(
+      // if the list set is empty, the string will be '' and thus deleted from IAVL
+      [...namespaces.keys()].sort().join(','),
+    );
+  };
+
+  return Far('prioritySenders manager', {
+    /**
+     * @param {string} namespace
+     * @param {string} address
+     * @returns {Promise<void>}
+     */
+    add: async (namespace, address) => {
```

Comment:
Ocap nit: I was imagining that each user would get its own facet with the namespace part of the readonly facet's state so that one user wouldn't be able to interfere with other users.

Reply:
acknowledged with XXX